frontend/src/components/Topics/Topic/Messages/MessagesTable.tsx

```diff
@@ -2,43 +2,81 @@ import PageLoader from 'components/common/PageLoader/PageLoader';
 import { Table } from 'components/common/table/Table/Table.styled';
 import TableHeaderCell from 'components/common/table/TableHeaderCell/TableHeaderCell';
 import { TopicMessage } from 'generated-sources';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Button } from 'components/common/Button/Button';
 import * as S from 'components/common/NewTable/Table.styled';
 import { usePaginateTopics, useIsLiveMode } from 'lib/hooks/useMessagesFilters';
 import { useMessageFiltersStore } from 'lib/hooks/useMessageFiltersStore';
+import useAppParams from 'lib/hooks/useAppParams';
+import { RouteParamsClusterTopic } from 'lib/paths';
+import { useLocalStorage } from 'lib/hooks/useLocalStorage';
 
-import PreviewModal from './PreviewModal';
 import Message, { PreviewFilter } from './Message';
+import PreviewModal from './PreviewModal';
 
 export interface MessagesTableProps {
   messages: TopicMessage[];
   isFetching: boolean;
 }
 
+interface MessagePreviewProps {
+  [key: string]: {
+    keyFilters: PreviewFilter[];
+    contentFilters: PreviewFilter[];
+  };
+}
+
 const MessagesTable: React.FC<MessagesTableProps> = ({
   messages,
   isFetching,
 }) => {
   const paginate = usePaginateTopics();
-  const [previewFor, setPreviewFor] = useState<string | null>(null);
-
+  const [previewFor, setPreviewFor] = useState<'key' | 'content' | null>(null);
   const [keyFilters, setKeyFilters] = useState<PreviewFilter[]>([]);
   const [contentFilters, setContentFilters] = useState<PreviewFilter[]>([]);
   const nextCursor = useMessageFiltersStore((state) => state.nextCursor);
   const isLive = useIsLiveMode();
+  const { topicName } = useAppParams<RouteParamsClusterTopic>();
+  const [messagesPreview, setMessagesPreview] =
+    useLocalStorage<MessagePreviewProps>('message-preview', {
+      [topicName]: {
+        keyFilters: [],
+        contentFilters: [],
+      },
+    });
+
+  useEffect(() => {
+    setKeyFilters(messagesPreview[topicName]?.keyFilters || []);
+    setContentFilters(messagesPreview[topicName]?.contentFilters || []);
+  }, []);
 
   return (
     <div style={{ position: 'relative' }}>
       {previewFor !== null && (
         <PreviewModal
           values={previewFor === 'key' ? keyFilters : contentFilters}
           toggleIsOpen={() => setPreviewFor(null)}
-          setFilters={(payload: PreviewFilter[]) =>
-            previewFor === 'key'
-              ? setKeyFilters(payload)
-              : setContentFilters(payload)
-          }
+          setFilters={(payload: PreviewFilter[]) => {
```

Comment:
Minor comment. I think it's better to write separate function for setting filters, for better readability.

Reply:
fixed